distribution/zip/jballerina/bin/bal

```diff
@@ -262,6 +262,27 @@ elif [[ $1 == "run" ]] && [[ $2 == *.jar ]]; then # if 1st arg is "run" and 2nd
         -Dballerina.target=jvm \
         -Djava.command=$JAVACMD \
         -jar "${@:2}" # ignores "run"
+
+elif [[ $1 == "run" ]] && [[ $2 == "--debug" ]]; then # if 1st arg is "run" and 2nd arg is "--debug".
```

Comment:
This logic will break the `bal run` command for ballerina package/single file, as the conventional CLI `run` commands (i.e. `bal run --debug 5005 <PACKAGE_NAME>` will be matched to this condition(hence the Ballerina CLI launcher will not be executed). Therefore move these debug agent args to the above `elif` condition.

Reply:
IMO we can not move the debug agent args to the above `elif` condition because the above `elif` condition runs the jar in normal mode. (i.e. `bal run <EXECUTABLE_JAR>`). Therefore adding an additional condition (`[[ $4 == *.jar ]`) here would work fine.
i.e. `elif [[ $1 == "run" ]] && [[ $2 == "--debug" ]] && [[ $4 == *.jar ]]; then`